aii-opennebula/src/main/perl/opennebula.pm

```diff
@@ -705,20 +708,20 @@ sub remove
         $self->stop_and_remove_one_vms($one,$fqdn);
     }
 
+    my $vmtemplatetxt = $self->get_vmtemplate($config, $oneversion);
+    if ($vmtemplatetxt) {
+        $self->remove_and_create_vm_template($one, $fqdn, 1, $vmtemplatetxt, undef, $rmvmtemplate);
```

Comment:
Should this be called when  $rmvmtemplate is false. The current behavior now is to update the template then when it is false (which I guess is not needed in a remove context)